src/SFA.DAS.AssessorService.Application.Api/Controllers/OrganisationSearchController.cs

```diff
@@ -0,0 +1,439 @@
+                        TradingName = group.Select(g => g.TradingName).FirstOrDefault(TradingName => !string.IsNullOrWhiteSpace(TradingName)),
+                        ProviderName = group.Select(g => g.ProviderName).FirstOrDefault(ProviderName => !string.IsNullOrWhiteSpace(ProviderName)),
+                        Address = group.Select(g => g.Address).FirstOrDefault(Address => Address != null),
+                        OrganisationType = group.Select(g => g.OrganisationType).FirstOrDefault(OrganisationType => !string.IsNullOrWhiteSpace(OrganisationType)),
+                        OrganisationReferenceType = group.Select(g => g.OrganisationReferenceType).FirstOrDefault(OrganisationReferenceType => !string.IsNullOrWhiteSpace(OrganisationReferenceType)),
+                        OrganisationReferenceId = string.Join(",", group.Select(g => g.OrganisationReferenceId).Where(Id => !string.IsNullOrWhiteSpace(Id)).Distinct()),
+                        RoEPAOApproved = group.Select(g => g.RoEPAOApproved).FirstOrDefault(RoEPAOApproved => RoEPAOApproved != false),
+                        RoATPApproved = group.Select(g => g.RoATPApproved).FirstOrDefault(RoATPApproved => RoATPApproved != false),
+                        CompanyNumber = group.Select(g => g.CompanyNumber).FirstOrDefault(CompanyNumber => !string.IsNullOrWhiteSpace(CompanyNumber)),
+                        CharityNumber = group.Select(g => g.CharityNumber).FirstOrDefault(CharityNumber => !string.IsNullOrWhiteSpace(CharityNumber)),
+                        EasApiOrganisationType = group.Select(g => g.EasApiOrganisationType).FirstOrDefault(EasApiOrganisationType => !string.IsNullOrWhiteSpace(EasApiOrganisationType)),
+                        FinancialDueDate = group.Select(g => g.FinancialDueDate).FirstOrDefault(FinancialDueDate => FinancialDueDate != null),
+                        FinancialExempt = group.Select(g => g.FinancialExempt).FirstOrDefault(FinancialExempt => FinancialExempt != null),
+                        OrganisationIsLive = group.Select(g => g.OrganisationIsLive).FirstOrDefault(OrganisationIsLive => OrganisationIsLive)
+                    }
+                );
+
+            return charityNumberMerge.OrderByDescending(org => org.Ukprn).ToList();
+        }
+
+       
+
+        [HttpGet("organisations/{companyNumber}/isActivelyTrading")]
+        public async Task<bool> isCompanyActivelyTrading(string companyNumber)
+        {
+
+            _logger.LogInformation($"isCompanyActivelyTrading({companyNumber})");
+
+            bool result = false;
+
+            // EPAO Register
+            try
+            {
+                result = await _companiesHouseApiClient.IsCompanyActivelyTrading(companyNumber);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Error from Companies House. Message: {ex.Message}");
+            }
+
+            return result;
+        }
+    }
+}
```

Comment:
Seems to be quite a lot going on in this Controller.  Can this be refactored out into the .Application / Handler etc?